src/dreyfus_index.erl

```diff
@@ -222,6 +258,22 @@ code_change(_OldVsn, State, _Extra) ->
 
 % private functions.
 
+get_index_with_sig(#doc{body={Fields}}=Doc, Sig) ->
+    RawIndexes = couch_util:get_value(<<"indexes">>, Fields, {[]}),
+    case RawIndexes of
+        {IndexList} when is_list(IndexList) ->
+            {IndexNames, _} = lists:unzip(IndexList),
+            lists:flatmap(
+                fun(IndexName) ->
+                    case (catch design_doc_to_index(Doc, IndexName)) of
+                        {ok, #index{sig=Sig_New}=Index} when Sig_New =:= Sig -> [Index];
+                        _ -> []
+                    end
+                end,
+                IndexNames);
+        _ -> []
```

Comment:
what else could it be? we should avoid defensive programming, it causes more problems than it solves.